src/mlpack/methods/ann/layer/dropout_impl.hpp

```diff
@@ -78,17 +78,53 @@ template<typename MatType>
 void DropoutType<MatType>::Forward(const MatType& input, MatType& output)
 {
   // The dropout mask will not be multiplied in testing mode.
+  ForwardImpl(input, output);
+}
+
+
+template<typename MatType>
+template<typename T, typename std::enable_if_t<arma::is_arma_type<T>::value, int>>
+void DropoutType<MatType>::ForwardImpl(const T& input, T& output)
+{
+  if (!this->training)
+  {
+    output = input;
+  }
+  else
+  {
+    mask.randu(input.n_rows, input.n_cols);
+    #pragma omp parallel for collapse(2)
+    for (size_t i = 0; i < input.n_rows; ++i)
+    {
+      for (size_t j = 0; j < input.n_cols; ++j)
+      {
+        mask(i, j) = (mask(i, j) > this->ratio) ? 1.0 : 0.0;
+      }
+    }
+    output = input % mask * this->scale;
+  }
+}
+
```

Comment:
@MarkFischinger It looks to me as if it is the same implementation when using Armadillo or Bandicoot.
It this is the case, then it is fine, no worries, I would just keep one of them, and remove SFINAE and the other overload since we no longer use the transform function.
Let me know if I overlooked something